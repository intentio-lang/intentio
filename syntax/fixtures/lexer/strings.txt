"Quick brown fox jumps over the lazy dog"

r"Quick brown fox jumps over the lazy dog"

xr"Quick brown fox jumps over the lazy dog"

c"a"

c"\u{1f436}"

"Quick brown fox
 jumps over
 the lazy dog"

r####"
Quick \u{1f43a} jumps over the lazy \u{1f436}
"####

tur####"
Quick \u{1f43a} jumps over the lazy \u{1f436}
"####

xr###"
Quick \u{1f43a} jumps over the lazy \u{1f436}
"###

# TODO: Tricky case, what to do here?
# r##"
# Quick \u{1f43a} jumps "####" the lazy \u{1f436}
# "##

"C:\\"

"\""
