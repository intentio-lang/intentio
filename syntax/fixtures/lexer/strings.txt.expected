[
  {
    "_text": "\"Quick brown fox jumps over the lazy dog\"",
    "_ty": "TString"
  },
  {
    "_text": "r\"Quick brown fox jumps over the lazy dog\"",
    "_ty": "TRawString"
  },
  {
    "_text": "xr\"Quick brown fox jumps over the lazy dog\"",
    "_ty": "TRegexString"
  },
  {
    "_text": "c\"a\"",
    "_ty": "TCharString"
  },
  {
    "_text": "c\"\\u{1f436}\"",
    "_ty": "TCharString"
  },
  {
    "_text": "\"Quick brown fox\n jumps over\n the lazy dog\"",
    "_ty": "TString"
  },
  {
    "_text": "r####\"\nQuick \\u{1f43a} jumps over the lazy \\u{1f436}\n\"####",
    "_ty": "TRawString"
  },
  {
    "_text": "tur####\"\nQuick \\u{1f43a} jumps over the lazy \\u{1f436}\n\"####",
    "_ty": "TRawString"
  },
  {
    "_text": "xr###\"\nQuick \\u{1f43a} jumps over the lazy \\u{1f436}\n\"###",
    "_ty": "TRegexString"
  },
  {
    "_text": "\"C:\\\\\"",
    "_ty": "TString"
  },
  {
    "_text": "\"\\\"\"",
    "_ty": "TString"
  }
]
